src/gui/tray/ActivityItemContent.qml

```diff
@@ -221,12 +221,12 @@ RowLayout {
 
         visible: root.activityData.isShareable
 
-        imageSource: "image://svgimage-custom-color/share.svg" + "/" + UserModel.currentUser.headerColor
+        imageSource: "image://svgimage-custom-color/share.svg" + "/" + themedHeaderColor
         imageSourceHover: "image://svgimage-custom-color/share.svg" + "/" + UserModel.currentUser.headerTextColor
 
         toolTipText: qsTr("Open share dialog")
 
-        bgColor: UserModel.currentUser.headerColor
+        bgColor: headerColor
```

Comment:
instead of that (because it may not be obvious from which place comes `headerColor`) you may want to declare an alias property
```
property alias headerColor: shareButton.bgColor
```

you would then assign it a value when the `ActivityItemContent` object is created in `ActivityItem` component

and please check if could make the other usage of `themedHeaderColor` more explicit in a similar way

Reply:
I went back and forth with it... it worked but then after a few glitches I decided to just use the Style to define these colors. Please take a look.